patternfly/patternfly-elements
# fix: update `engines.node` for elements and monorepo root

## <Pull request description>

This PR changes the `engines.node` setting in each elements' package.json, as well as the same setting in the root package.json, to remove unintended restrictions on where PatternFly Elements are used, while trying to preserve the restrictions' purpose when it comes to development tooling.

### Related issues

- (#1744) `engines` blocks yarn install, adds warning to npm install

### What has changed and why



1. `engines.node` setting removed from all elements
    - Why: so PFE's can be installed from npm without triggering the issues described in #1744
2. `engines.node` setting removed from the generator
    - Why: so yet-to-be-created PFE's can be installed from npm without triggering the issues described in #1744
3. `engines.node` at the root changed from `>=10 <13` to `>=12 <16`
    - Why: `engines` at the root can provide a warning to contributors if they're using an unsupported version of node to build the project.  For instance, node 16 currently fails to build the project.
4. node 14 added to CI tests
    - Why: to detect problems related to building in different node versions, and maintain support for the node versions listed in package.json.


### Testing instructions



- [x] *Test for npm warning*
  1. `nvm use 16`
  2. `npm install elements/pfe-cta`
  3. No warning should be emitted about node version
- [x] *Test for yarn error*
  1. `nvm use 16`
  2. `npx yarn add elements/pfe-cta`
  3. No error should be emitted about node version


### Ready-for-merge Checklist




### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
